WordPress/gutenberg
# BorderControl: Update labelling, tooltips and wrap with fieldset and legend

Related: 
- https://github.com/WordPress/gutenberg/issues/42095

## What?

Updates the `BorderControl` component to improve labelling, DOM structure, and tooltips.

_Note: Improvements to the `BorderBoxControl` will be addressed in a separate PR_

## Why?

Accessibility is important.

## How?
- Changes the `BorderControl` wrapping element to a `fieldset` and its primary label to a `legend`
- Adds a label to the inner `UnitControl` and `RangeControl` components
- Adds a Tooltip to the "Border color and style picker" dropdown button
- Updates the `getByRole` and `queryByRole` calls in unit tests to also filter by name

## Testing Instructions
1. Run `npm run test-unit /packages/components/src/border-control/test`
2. Fire up the Storybook and test the `BorderControl` still functions correctly
    - http://localhost:50240/?path=/story/components-experimental-bordercontrol--default
3. Inspect the border control component and its inner inputs confirming appropriate labels
4. Hover over the border color/style dropdown button and confirm tooltip displays
